lib/pbench/server/cache_manager.py

```diff
@@ -1507,6 +1521,58 @@ def _discover_controllers(self):
             if file.is_dir() and file.name != CacheManager.TEMPORARY:
                 self._add_controller(file)
 
+    def _discover_datasets(self):
+        """Discover the ARCHIVE tree from the SQL representation of datasets
+
+        Discover all controllers and tarballs with server.tarball-path
+        metadata. This will find only correct and "live" tarballs, and may
+        miss incorrectly labeled tarballs. (Unless we're trying to diagnose
+        ARCHIVE tree errors, this is probably what we want.)
+        """
+        rows = (
+            Database.db_session.query(
+                Dataset.name,
+                Dataset.resource_id,
+                Metadata.value["tarball-path"].as_string(),
+            )
+            .execution_options(stream_results=True)
+            .outerjoin(
+                Metadata,
+                and_(Dataset.id == Metadata.dataset_ref, Metadata.key == "server"),
+            )
+            .yield_per(500)
+            .all()
```

Comment:
Did you look at using a number larger than `500`?  Given the size of a "row" and the number of rows that there might be, it seems like 10 or 20 times that many would be better.

Reply:
I used these parameters to stream a query in the report generator earlier, and I just copied them. The intention is to avoid overloading Python's heap with hundreds of thousands of rows at once, but I really did't want to get into trying to figure out how to get good enough low-level usage data to seriously optimize it. Yeah, it's possible there'd even be a visible difference at the CLI level, which is the only place this is used: but I'm not sure that investigation is a particularly good use of my time right now.

Reply:
> I'm not sure that investigation is a particularly good use of my time right now.

Prob'ly not...I was just thinking that 130K results taken 500 at a time is a lot of takes, especially since I suspect that each block of 500 doesn't take up much space in real terms.